### ISSUE TITLE
crash when importing on 3.6.8 on debian stretch

### ISSUE THREAD

**Issue by thehesiod** [dvarrazzo/py-setproctitle#62]
We've been using this for several years w/o any issues, but just last friday it started crashing.

```bash
bash-4.4# pip3 install -v setproctitle
Created temporary directory: /tmp/pip-ephem-wheel-cache-1plph2bo
Created temporary directory: /tmp/pip-req-tracker-bcnloap1
Created requirements tracker '/tmp/pip-req-tracker-bcnloap1'
Created temporary directory: /tmp/pip-install-ji72t5kg
Looking in indexes: https://pypi.fbn.org/simple
Collecting setproctitle
  1 location(s) to search for versions of setproctitle:
  * https://pypi.fbn.org/simple/setproctitle/
  Getting page https://pypi.fbn.org/simple/setproctitle/
  Starting new HTTPS connection (1): pypi.fbn.org:443
  https://pypi.fbn.org:443 "GET /simple/setproctitle/ HTTP/1.1" 302 212
  Starting new HTTPS connection (1): pypi.python.org:443
  https://pypi.python.org:443 "GET /simple/setproctitle/ HTTP/1.1" 301 122
  Starting new HTTPS connection (1): pypi.org:443
  https://pypi.org:443 "GET /simple/setproctitle/ HTTP/1.1" 200 2960
  Analyzing links from page https://pypi.org/simple/setproctitle/
...
  Using version 1.1.10 (newest of versions: 0.1, 0.2, 0.3, 0.4, 1.0, 1.0.1, 1.1, 1.1.1, 1.1.2, 1.1.3, 1.1.4, 1.1.5, 1.1.6, 1.1.7, 1.1.8, 1.1.9, 1.1.10)
  Created temporary directory: /tmp/pip-unpack-cyoon6f4
  Starting new HTTPS connection (1): files.pythonhosted.org:443
  https://files.pythonhosted.org:443 "GET /packages/5a/0d/dc0d2234aacba6cf1a729964383e3452c52096dc695581248b548786f2b3/setproctitle-1.1.10.tar.gz HTTP/1.1" 200 24042
  Downloading https://files.pythonhosted.org/packages/5a/0d/dc0d2234aacba6cf1a729964383e3452c52096dc695581248b548786f2b3/setproctitle-1.1.10.tar.gz
  Downloading from URL https://files.pythonhosted.org/packages/5a/0d/dc0d2234aacba6cf1a729964383e3452c52096dc695581248b548786f2b3/setproctitle-1.1.10.tar.gz#sha256=6283b7a58477dd8478fbb9e76defb37968ee4ba47b05ec1c053cb39638bd7398 (from https://pypi.org/simple/setproctitle/)
  Added setproctitle from https://files.pythonhosted.org/packages/5a/0d/dc0d2234aacba6cf1a729964383e3452c52096dc695581248b548786f2b3/setproctitle-1.1.10.tar.gz#sha256=6283b7a58477dd8478fbb9e76defb37968ee4ba47b05ec1c053cb39638bd7398 to build tracker '/tmp/pip-req-tracker-bcnloap1'
    Running setup.py (path:/tmp/pip-install-ji72t5kg/setproctitle/setup.py) egg_info for package setproctitle
    Running command python setup.py egg_info
    running egg_info
    creating pip-egg-info/setproctitle.egg-info
    writing pip-egg-info/setproctitle.egg-info/PKG-INFO
    writing dependency_links to pip-egg-info/setproctitle.egg-info/dependency_links.txt
    writing top-level names to pip-egg-info/setproctitle.egg-info/top_level.txt
    writing manifest file 'pip-egg-info/setproctitle.egg-info/SOURCES.txt'
    reading manifest file 'pip-egg-info/setproctitle.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'pip-egg-info/setproctitle.egg-info/SOURCES.txt'
  Source in /tmp/pip-install-ji72t5kg/setproctitle has version 1.1.10, which satisfies requirement setproctitle from https://files.pythonhosted.org/packages/5a/0d/dc0d2234aacba6cf1a729964383e3452c52096dc695581248b548786f2b3/setproctitle-1.1.10.tar.gz#sha256=6283b7a58477dd8478fbb9e76defb37968ee4ba47b05ec1c053cb39638bd7398
  Removed setproctitle from https://files.pythonhosted.org/packages/5a/0d/dc0d2234aacba6cf1a729964383e3452c52096dc695581248b548786f2b3/setproctitle-1.1.10.tar.gz#sha256=6283b7a58477dd8478fbb9e76defb37968ee4ba47b05ec1c053cb39638bd7398 from build tracker '/tmp/pip-req-tracker-bcnloap1'
Installing collected packages: setproctitle
  Created temporary directory: /tmp/pip-record-nigsrwna
  Running setup.py install for setproctitle ...     Running command /usr/local/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-ji72t5kg/setproctitle/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-nigsrwna/install-record.txt --single-version-externally-managed --compile
    running install
    running build
    running build_ext
    building 'setproctitle' extension
    creating build
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -flto -fuse-linker-plugin -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -std=c99 -g -O2 -fdebug-prefix-map=/build/Python-3.6.8=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DSPT_VERSION=1.1.10 -DHAVE_SYS_PRCTL_H=1 -I/usr/local/include/python3.6m -c src/setproctitle.c -o build/temp.linux-x86_64-3.6/src/setproctitle.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -flto -fuse-linker-plugin -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -std=c99 -g -O2 -fdebug-prefix-map=/build/Python-3.6.8=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DSPT_VERSION=1.1.10 -DHAVE_SYS_PRCTL_H=1 -I/usr/local/include/python3.6m -c src/spt_debug.c -o build/temp.linux-x86_64-3.6/src/spt_debug.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -flto -fuse-linker-plugin -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -std=c99 -g -O2 -fdebug-prefix-map=/build/Python-3.6.8=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DSPT_VERSION=1.1.10 -DHAVE_SYS_PRCTL_H=1 -I/usr/local/include/python3.6m -c src/spt_setup.c -o build/temp.linux-x86_64-3.6/src/spt_setup.o
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -flto -fuse-linker-plugin -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -std=c99 -g -O2 -fdebug-prefix-map=/build/Python-3.6.8=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DSPT_VERSION=1.1.10 -DHAVE_SYS_PRCTL_H=1 -I/usr/local/include/python3.6m -c src/spt_status.c -o build/temp.linux-x86_64-3.6/src/spt_status.o
    src/spt_status.c: In function ‘save_ps_display_args’:
    src/spt_status.c:221:38: warning: implicit declaration of function ‘strdup’ [-Wimplicit-function-declaration]
                         new_environ[i] = strdup(environ[i]);
                                          ^~~~~~
    src/spt_status.c:221:36: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
                         new_environ[i] = strdup(environ[i]);
                                        ^
    src/spt_status.c:253:25: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
                 new_argv[i] = strdup(argv[i]);
                             ^
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -g -flto -fuse-linker-plugin -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -std=c99 -g -O2 -fdebug-prefix-map=/build/Python-3.6.8=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -DSPT_VERSION=1.1.10 -DHAVE_SYS_PRCTL_H=1 -I/usr/local/include/python3.6m -c src/spt_strlcpy.c -o build/temp.linux-x86_64-3.6/src/spt_strlcpy.o
    creating build/lib.linux-x86_64-3.6
    gcc -pthread -shared -Wl,-z,relro -Wl,-z,relro build/temp.linux-x86_64-3.6/src/setproctitle.o build/temp.linux-x86_64-3.6/src/spt_debug.o build/temp.linux-x86_64-3.6/src/spt_setup.o build/temp.linux-x86_64-3.6/src/spt_status.o build/temp.linux-x86_64-3.6/src/spt_strlcpy.o -L/usr/local/lib -lpython3.6m -o build/lib.linux-x86_64-3.6/setproctitle.cpython-36m-x86_64-linux-gnu.so
    running install_lib
    copying build/lib.linux-x86_64-3.6/setproctitle.cpython-36m-x86_64-linux-gnu.so -> /usr/local/lib/python3.6/site-packages
    running install_egg_info
    running egg_info
    writing setproctitle.egg-info/PKG-INFO
    writing dependency_links to setproctitle.egg-info/dependency_links.txt
    writing top-level names to setproctitle.egg-info/top_level.txt
    reading manifest file 'setproctitle.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'setproctitle.egg-info/SOURCES.txt'
    Copying setproctitle.egg-info to /usr/local/lib/python3.6/site-packages/setproctitle-1.1.10-py3.6.egg-info
    running install_scripts
    writing list of installed files to '/tmp/pip-record-nigsrwna/install-record.txt'
done
  Removing source in /tmp/pip-install-ji72t5kg/setproctitle
Successfully installed setproctitle-1.1.10
Cleaning up...
Removed build tracker '/tmp/pip-req-tracker-bcnloap1'
1 location(s) to search for versions of pip:
* https://pypi.fbn.org/simple/pip/
Getting page https://pypi.fbn.org/simple/pip/
Starting new HTTPS connection (1): pypi.fbn.org:443
https://pypi.fbn.org:443 "GET /simple/pip/ HTTP/1.1" 302 203
Starting new HTTPS connection (1): pypi.python.org:443
https://pypi.python.org:443 "GET /simple/pip/ HTTP/1.1" 301 122
Starting new HTTPS connection (1): pypi.org:443
https://pypi.org:443 "GET /simple/pip/ HTTP/1.1" 200 10884
Analyzing links from page https://pypi.org/simple/pip/
....
bash-4.4# python3
Python 3.6.8 (default, Apr  5 2019, 22:35:05) 
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import setproctitle
Segmentation fault

```

**Comment by thehesiod** [comment]
seems like this may be our fault, just found out that between two builds of python 3.6.8 the issue started happening.

**Comment by thehesiod** [comment]
btw, it appears this was caused by adding `-std=c99` when compiling python

**Comment by thehesiod** [comment]
actually, given only this module crashes, going to re-open, shouldn't crash with `-std=c99` compile line

**Comment by dvarrazzo** [comment]
As you are playing with it, can you maybe provide a patch?

**Comment by thehesiod** [comment]
I have a docker container that can reproduce the crash, a specific test you want me run?

**Comment by thehesiod** [comment]
btw this whole mess was triggered by: https://bugs.python.org/issue36478  put some info there

**Comment by dvarrazzo** [comment]
Assuming fixed merging #70. If that's not the case please reopen and provide a test.
